admin/admin_page_registration.php

```diff
@@ -5,91 +5,88 @@
  * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
  * @version $Id: DrByte 2020 Jun 07 Modified in v1.5.7 $
  */
-require('includes/application_top.php');
+require 'includes/application_top.php';
 
 // initialise form values
 $page_key = $language_key = $main_page = $page_params = $menu_key = $checked = '';
 $sort_order = 0;
 
 // check if we are receiving form content and if so validate and process it
 if (isset($_POST) && !empty($_POST)) {
-  $error = FALSE;
+    $error = FALSE;
 
-  if (isset($_POST['page_key'])) {
-    $page_key = zen_db_prepare_input($_POST['page_key']);
-  }
-  if (empty($page_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_PAGE_KEY_NOT_ENTERED, 'error');
-  } elseif (zen_page_key_exists($page_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_PAGE_KEY_ALREADY_EXISTS, 'error');
-  }
+    $keys = [
+        'page_key' => [
+            'empty' => true,
+            'elseif' => 'zen_page_key_exists',
+        ],
+        'language_key' => [
+            'empty' => true,
+            'elseif' => 'defined',
+            'elseifNot' => true,
+        ],
+        'main_page' => [
+            'empty' => true,
+            'elseif' => 'defined',
+            'elseifNot' => true,
+        ],
+        'page_params',
+        'menu_key' => [
+            'empty' => true,
+        ],
+        'sort_order' => [
+            'valueType' => 'integer',
+        ],
+    ];
 
-  if (isset($_POST['language_key'])) {
-    $language_key = zen_db_prepare_input($_POST['language_key']);
-  }
-  if (empty($language_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_LANGUAGE_KEY_NOT_ENTERED, 'error');
-  } elseif (!defined($language_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_LANGUAGE_KEY_HAS_NOT_BEEN_DEFINED, 'error');
-  }
+    foreach ($keys as $key => $value) {
+        if (isset($_POST[$key])) {
+            if (!empty($value['valueType'])) {
+                ${$key} = $db->bindVars(':val:', ':val:', $_POST[$key], $value['valueType']);

```

Comment:
Can't we just cast to integer if $key = sort_order?

Reply:
Sorry for delay in response, other browser wouldn't let me write in the "code review" type box.

Was trying to use/develop code that was independent of some sort of specific "value" or setting.. In a way I would prefer all of the keys to have a `valueType` and just process it/them to the associated value noting that at this point in execution the data is expected to have already gone through sanitization (to whatever level the store has active).